improve print_model_weights output by adding a new argument to sort feature weights by sort_by_labels #442

bndgyawali opened the pull request:
This PR improves `print_model_weights` by adding a new argument `--sort_by_labels` which when true sorts the features by classes. By default, it is false. 

This new argument is mutually exclusive to `--k` which mean it only works when `--k` is not specified.

bndgyawali commented:
I am going to add a new test case. So, please hold a while to review it.

desilinguist commented on skll/utilities/print_model_weights.py:
```diff
@@ -58,6 +64,13 @@ def main(argv=None):
+        isinstance(model, LogisticRegression) or
+        (isinstance(model, SVC) and
+            model.kernel == 'linear')):
+        multiclass = True
```
You could be using `LogisticRegression` for bianary classification in which case you will only have weights for one class and then you don't need to do this. May be you need an condition that the number of classes (via `self.label_list`) is  > 2?

desilinguist commented on skll/utilities/print_model_weights.py:
```diff
@@ -69,19 +82,28 @@ def main(argv=None):
+            label, feature = label_feature.split()
+            weight_by_class[label][feature] = weight
+        for label in sorted(weight_by_class):
+            for feat, val in sorted(weight_by_class[label].items(), key=lambda x: -abs(x[1])):
```
You need to show a warning to the suer that the `k` part is ignored when you are sorting.

bndgyawali commented:
I have addressed the comments and added extra tests to test the output of `print_model_weights` when `sort_by_labels` is passed as an argument. 
And, it's ready for review.

desilinguist commented:
@jbiggsets @mulhod if you get a chance, please review this PR.
